week-2/Homework/extra/2-array-methods.js

```diff
@@ -5,31 +5,72 @@ Following are set of small functions that you need to create using loops and arr
 //1)Find the Smallest Number in an Array
 //Create a function that takes an array of numbers and return the smallest number in it
 
-function findSmallestNum(arr) {}
+function findSmallestNum(arr) {
+    let min = arr[0];
+    
+    for ( i = 1; i <=arr.length - 1;i ++ ){
+        if (arr[i] < min ){
+            min = arr[i];
+        }   
+    }
+    return min
+}
+// console.log(findSmallestNum([354, 15, 114, 2]));
 
 findSmallestNum([354, 15, 114, 2]); //Expected: 2
 
 findSmallestNum([-76, 1, -79, 1, 0]); //Expected: -79
 
 //2)Checking Even Numbers
 //Create a function that takes in an array and returns true if all its values are even, and false otherwise.
-function checkAllEven(arr) {}
+function checkAllEven(arr) {
+    let result = true;
+    for ( i in arr ) {
+        if (!(arr[i] % 2 == 0)){
+            result = false;

```

Comment:
Not sure if you've already learnt it or not, it might be advanced tip at this point but
You can immediately exit the loop using `return false` statement once we know array doesn't consist of only pure even numbers. When you use `return` it will interrupt and leave the function computation at whichever point it was. (You can put as many return points into your function as you want.) In this case, at the end of this function, you can always say `return true` because we can reach that point if and only if the loop hasn't ended the function call by returning a value earlier.